cli/cmd/report_rules.go

```diff
@@ -0,0 +1,399 @@
+		resourceGroups := [][]string{{strings.Join(rule.Filter.ResourceGroups, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"RESOURCE GROUPS"}, resourceGroups,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+	}
+
+	return detailsTable.String()
+}
+
+func promptCreateReportRule() (api.ReportRuleResponse, error) {
+	channelList, channelMap := getEmailAlertChannels()
+	notificationFields := structs.Names(api.ReportRuleNotificationTypes{})
+	notificationsMap := make(map[string]bool)
+
+	if len(channelList) < 1 {
+		return api.ReportRuleResponse{}, errors.New("no email alert channels found.")
+	}
+
+	questions := []*survey.Question{
+		{
+			Name:     "name",
+			Prompt:   &survey.Input{Message: "Name: "},
+			Validate: survey.Required,
+		},
+		{
+			Name:     "description",
+			Prompt:   &survey.Input{Message: "Description: "},
+			Validate: survey.Required,
+		},
+		{
+			Name: "channels",
+			Prompt: &survey.MultiSelect{
+				Message: "Select email alert channels:",
+				Options: channelList,
+			},
+			Validate: survey.Required,
+		},
+		{
+			Name: "severities",
+			Prompt: &survey.MultiSelect{
+				Message: "Select severities:",
+				Options: []string{"Critical", "High", "Medium", "Low", "Info"},
+			},
+		},
+		{
+			Name: "notifications",
+			Prompt: &survey.MultiSelect{
+				Message: "Select report notification types:",
+				Options: notificationFields,
+			},
+		},
+	}
+
+	answers := struct {
+		Name           string
+		Description    string   `survey:"description"`
+		Channels       []string `survey:"channels"`
+		Severities     []string `survey:"severities"`
+		ResourceGroups []string `survey:"resourceGroups"`
+		Notifications  []string `survey:"notifications"`
+	}{}
+
+	err := survey.Ask(questions, &answers,
+		survey.WithIcons(promptIconsFunc),
+	)
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	var channels []string
+	for _, channel := range answers.Channels {
+		channels = append(channels, channelMap[channel])
+	}
+
+	resourceGroups, resourceGroupMap := promptAddResourceGroupsToReportRule()
+	var groups []string
+	for _, group := range resourceGroups {
+		groups = append(groups, resourceGroupMap[group])
+	}
+
+	for _, n := range answers.Notifications {
+		notificationsMap[n] = true
+	}
+
+	notifications := api.ReportRuleNotificationTypes{}
+	err = api.TransformReportRuleNotification(notificationsMap, &notifications)
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	reportRule, err := api.NewReportRule(
+		answers.Name,
+		api.ReportRuleConfig{
+			Description:        answers.Description,
+			Severities:         api.NewReportRuleSeverities(answers.Severities),
+			ResourceGroups:     groups,
+			EmailAlertChannels: channels,
+			NotificationTypes:  api.ReportRuleNotifications{notifications},
+		})
+
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	cli.StartProgress(" Creating report rule...")
```

Comment:
I really like we use the progress widget, can we please add it to other commands? (`show`, `delete`, and `list`)

Reply:
Done